kds/src/commonMain/kotlin/com/soywiz/kds/Array2.kt

```diff
@@ -241,14 +290,12 @@ data class IntArray2(val width: Int, val height: Int, val data: IntArray) : Iter
         }
     }
 
-    fun set(rows: List<List<Int>>) {
-        var n = 0
-        for (y in rows.indices) {
-            val row = rows[y]
-            for (x in row.indices) {
-                this.data[n++] = row[x]
-            }
-        }
+    override fun set(idx: Int, value: Int) {
+        this.data[idx] = value
+    }
```

Comment:
This would incur in boxing, and that would break the whole point of having an specialized class. Please check: https://github.com/korlibs/korge-next/commit/86ec6e0cc08d629d2a92aeaf23c849e776f44cad#diff-528f00b875719471f92b82e3e23e9d79bda46d3b03427de315cd6a863f4fa261L60-L61

So what I would suggest is to have methods like `getAt` and `setAt` that are generic, but the specialized one is not in the interface, to prevent a regression in performance.

What do you think=

Reply:
SGTM